php-frontend/src/main/java/org/sonar/php/tree/symbols/SymbolTableImpl.java

```diff
@@ -100,6 +106,7 @@ public Scope getScopeFor(Tree tree) {
     return scopes.get(tree);
   }
 
+  @CheckForNull
   @Override
   public Framework getFramework() {
     return this.framework;
```

Comment:
Maybe framework should be initialized to `EMPTY` instead? I don't expect differences in rule behavior whether it's detected as EMPTY or unset.